File: java-api/src/main/java/de/davelee/personalman/api/AddTrainingRequest.java
package de.davelee.personalman.api;

import lombok.*;

/**
 * This class is part of the PersonalMan REST API. It represents a request to add a new training course or qualification
 * to the user's profile.
 * @author Dave Lee
 */
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Setter
public class AddTrainingRequest {

    //company that the person works for
    private String company;

    //username of the person
    private String username;

    //token of the user making the change
    private String token;

    //name of the training course or qualification to be added to the user's profile
    private String trainingCourse;

}

File: server/src/users/requests/addtraining.request.ts
import { ApiProperty } from "@nestjs/swagger";

export class AddTrainingRequest {

    //company that the person works for
    @ApiProperty()
    private company: string;

    //username of the person
    @ApiProperty()
    private username: string;

    //token of the user making the change
    @ApiProperty()
    private token: string;

    //name of the training course or qualification to be added to the user's profile
    @ApiProperty()
    private trainingCourse: string;

    getCompany(): string {
        return this.company;
    }

    getUsername(): string {
        return this.username;
    }

    getToken(): string {
        return this.token;
    }

    getTrainingCourse(): string {
        return this.trainingCourse;
    }

}